src/lib/statistics/statistics_objects/range_filter.cpp

```diff
@@ -130,30 +130,30 @@ std::unique_ptr<RangeFilter<T>> RangeFilter<T>::build_filter(const pmr_vector<T>
   static_assert(std::is_arithmetic_v<T>, "Range filters are only allowed on arithmetic types.");
 
   DebugAssert(max_ranges_count > 0, "Number of ranges to create needs to be larger zero.");
-  DebugAssert(std::is_sorted(dictionary.begin(), dictionary.cend()), "Dictionary must be sorted in ascending order.");
 
   if (dictionary.empty()) {
     // Empty dictionaries will, e.g., occur in segments with only NULLs - or empty segments.
     return nullptr;
   }
 
   if (dictionary.size() == 1) {
-    std::vector<std::pair<T, T>> ranges;
-    ranges.emplace_back(dictionary.front(), dictionary.front());
+    auto ranges = std::vector<std::pair<T, T>>{{dictionary.front(), dictionary.front()}};
     return std::make_unique<RangeFilter<T>>(std::move(ranges));
   }
 
-  /*
-  * The code to determine the range boundaries requires calculating the distances between values. We cannot express the
-  * distance between -DBL_MAX and DBL_MAX using any of the standard data types. For these cases, the RangeFilter
-  * effectively degrades to a MinMaxFilter (i.e., stores only a single range).
-  * While being rather unlikely for doubles, it's more likely to happen when Hyrise includes tinyint etc.
-  * std::make_unsigned<T>::type would be possible to use for signed int types, but not for floating types.
-  * Approach: take the min and max values and simply check if the distance between both might overflow.
-  */
-  const auto min_max = std::minmax_element(dictionary.cbegin(), dictionary.cend());
-  if ((*min_max.first < 0) && (*min_max.second > std::numeric_limits<T>::max() + *min_max.first)) {
-    return std::make_unique<RangeFilter<T>>(std::vector<std::pair<T, T>>{{*min_max.first, *min_max.second}});
+  /**
+   * The code to determine the range boundaries requires calculating the distances between values. We cannot express the
+   * distance between -DBL_MAX and DBL_MAX using any of the standard data types. For these cases, the RangeFilter
+   * effectively degrades to a MinMaxFilter (i.e., stores only a single range).
+   * While being rather unlikely for doubles, it's more likely to happen when Hyrise includes tinyint etc.
+   * std::make_unsigned<T>::type would be possible to use for signed int types, but not for floating types.
+   * Approach: take the min and max values and simply check if the distance between both might overflow.
+   */
+  DebugAssert(std::is_sorted(dictionary.cbegin(), dictionary.cend()), "Dictionary must be sorted in ascending order.");
```

Comment:
I'd assume it's already checked before that the dict is sorted?

Reply:
Not necessarily here: Called from the following function:
https://github.com/hyrise/hyrise/blob/f96c18f3791ea82499360c11a44f442b34390147/src/lib/statistics/generate_pruning_statistics.cpp#L26-L38

The dictionary can either be the actual dictionary of a segment or is created on the fly:
https://github.com/hyrise/hyrise/blob/f96c18f3791ea82499360c11a44f442b34390147/src/lib/statistics/generate_pruning_statistics.cpp#L26-L38

So no assertion before.